ignite/metrics/precision.py

```diff
@@ -46,24 +65,33 @@ def compute(self) -> Union[torch.Tensor, float]:
                 f"{self.__class__.__name__} must have at least one example before it can be computed."
             )
         if not self._is_reduced:
-            if not (self._type == "multilabel" and not self._average):
+            if self._average == "samples":
+                self._sum_samples_metric = idist.all_reduce(self._sum_samples_metric)  # type: ignore[assignment]
+                self._samples_cnt = idist.all_reduce(self._samples_cnt)  # type: ignore[assignment]
+            else:
                 self._true_positives = idist.all_reduce(self._true_positives)  # type: ignore[assignment]
                 self._positives = idist.all_reduce(self._positives)  # type: ignore[assignment]
-            else:
-                self._true_positives = cast(torch.Tensor, idist.all_gather(self._true_positives))
-                self._positives = cast(torch.Tensor, idist.all_gather(self._positives))
+            if self._average == "weighted":
+                self._actual_positives = idist.all_reduce(self._actual_positives)  # type: ignore[assignment]
             self._is_reduced = True  # type: bool
 
-        result = self._true_positives / (self._positives + self.eps)
+        if self._average == "samples":
+            return (self._sum_samples_metric / self._samples_cnt).item()  # type: ignore
 
-        if self._average:
-            return cast(torch.Tensor, result).mean().item()
+        result = self._true_positives / (self._positives + self.eps)
+        if self._average == "weighted":
+            denominator = self._actual_positives.sum() + self.eps  # type: ignore
+            return ((result @ self._actual_positives) / denominator).item()  # type: ignore
+        elif self._average == "micro":
+            return result.item()  # type: ignore
+        elif self._average == "macro":
+            return result.mean().item()  # type: ignore
```

Comment:
Why we have to `type: ignore` ?

Reply:
Because we have
```python
result = self._true_positives / (self._positives + self.eps)
```
and `_true_positives` and `_positives` are of typehint `Union[int, torch.Tensor]` . So `result`'s type would implicitly be `Union[float, torch.Tensor]`. In line 88 we call `mean()` on `result`. On the other hand each feasible type of `result` should have `mean` method, but float has not. So MyPy raises error if I do not add the hint